src/journal.rs

```diff
@@ -66,7 +60,8 @@ pub type JournalRecord = BTreeMap<String, String>;
 ///
 /// Supports read, next, previous, and seek operations.
 pub struct Journal {
-    j: ffi::sd_journal
+    j: ffi::sd_journal,
+    cursor: String,
```

Comment:
Why do we need to cache a copy of the cursor here? Why not just have an accessor that retrieves it via ffi?